autogen/main/variables.tf.tmpl

```diff
@@ -387,8 +387,15 @@ variable "master_ipv4_cidr_block" {
 {% if beta_cluster %}
 
 variable "istio" {
-  description = "(Beta) Enable Istio addon"
-  default     = false
+  description = "Istio configs"
+  type        = object({
```

Comment:
I'd like to avoid changing this variable type. Instead can we add a new variable `istio_auth` and leave the existing setting untouched?

Reply:
@morgante my reasoning for using the object type was to keep it aligned with the way we are using `cluster_autoscaling` as we may have new arguments come in. But happy to change it if necessary.

Reply:
Since doing defaults for objects is still complicated, I'd prefer to keep this flat (with `var.istio ` and `var.istio_auth`) for now.

Reply:
gotcha, sg

Reply:
updated to use istio_auth